### PR TITLE
Fix docsite publish

### PR DESCRIPTION
**Pull request by DaniruKun**
This PR fixes the docsite publishing, which was previously missing uploading and downloading of the docsite static files artefact.

### CODE REVIEW

**Comment by axelson** [comment]
@DaniruKun Thanks for this! Can you temporarily cause this to run on every PR? I want ensure the site gets built and I don't want to make another tag just to get the site built and deployed. Once it is built and deployed we can change it back to run only on tags and then merge this PR.

**Comment by DaniruKun** [comment]
@axelson please make sure the `mkdocs` branch exists

File: .github/workflows/docsite.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        uses: actions/download-artifact@v1
+        with:
+          name: site
+          path: site
+      - name: Publish
+        uses: JamesIves/github-pages-deploy-action@4.1.7
+        with:
+          branch: mkdocs
+          folder: site
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (axelson) [inline .github/workflows/docsite.yml:49]: Judging by the documentation the token shouldn't need to be passed directly, so we should be okay if we just remove this line.

```suggestion
```

**Comment by axelson** [comment]
Okay, in https://github.com/elixir-lsp/elixir-ls/pull/640 I've deployed the initial version of the site :tada: 
https://elixir-lsp.github.io/elixir-ls/
I'm going to close 640 so now we can set this back to build on tags. Then this should be good to merge!

**Comment by DaniruKun** [comment]
@axelson so no changes are needed in the workflow file, apart from trigger on new tag push on master?

**Comment by axelson** [comment]
@DaniruKun yes that's correct, but it should be built when there's a tag, not from pushes to master. The issue is that the build previously was being triggered from a PR, where the branch that backed the PR was from a fork, which meant that the token associated with the PR did not have the appropriate permissions. Now that we'll be triggering the workflow from a tag that shouldn't be an issue.

**Comment by axelson** [comment]
Okay, I changed it to build on tags in 00b7ddef2445895497f2c1a43a672efb31150c43

**Review by axelson — APPROVED** [review]
Thank you! :heart:

**Pull request merged**